### ISSUE TITLE
[BUG] Install Script Abends at Line 1484 (system restart nginx)

### ISSUE THREAD

**Issue by ntsux** [Zer0CoolX/guacamole-install-rhel#128]
**Confirmation of Process**
1. You have read the [How to report issues](https://github.com/Zer0CoolX/guacamole-install-rhel/wiki/How-to-Report-Issues-(Bugs,-Feature-Request-and-Help)) wiki page? Y/N
2. You have read the wiki and followed its guidelines including requirements, troubleshooting, etc.? Y/N

Yes

**Details**
OS/Version: CentOS 7.7 via .iso file,  (VM on ESXi)
Guacamole Source: Stable/Git? Stable
Any other options selected from the script prompts:

- branding.jar extension
- Let's encrypt
- local DB for primary auth
- MFA for secondary auth, with sha 512

- All other settings, using defaults for all components/prompts

**Describe the bug**

The guac-install abends during Nginx configuration stage, while trying to restart the nginx service.  Line 1484 of the current stable script.

**Error Message(s) and Logs (be sure not to include sensitive/personal information!)**
If an error occurred include the exact message from the display/log.

Error Message:

%%% ERROR (Script Failed) | Line 1484 | Command: systemctl restart nginx | Exit code: 1 %%%

**To Reproduce**
Steps to reproduce the behavior (example):

4. See error (error message above)

**Have You Found a Solution to the Issue?**
Either by manually doing something before/after the script has run or by editing the script prior to running.  

Nope.  Wish I was better at shell scripting to help out with this!  I tried my best to reverse-engineer for a fix, but I just don't have the chops...

This looks very similar to issue #50 .   Just not exactly sure how it was resolved (my ignorance).  Since that was closed, I am hoping it's appropriate to open this new issue.

*** By the way, this is one of the finest scripts I have ever seen, very well structured, with very well laid out documentation!  Really appreciate your efforts here! ***

**Comment by ntsux** [comment]
Snip from logfile during abend:
-----------------------------------
Nginx Configuration

-Making Nginx config backup...    

-Generate Nginx guacamole.config...    

-Generate Nginx guacamole_ssl.config...    

-Hardening Nginx config...    
Created symlink from /etc/systemd/system/multi-user.target.wants/nginx.service to /usr/lib/systemd/system/nginx.service.

-Enable & Start Nginx Service...    
Job for nginx.service failed because the control process exited with error code. See "systemctl status nginx.service" and "journalctl -xe" for details.


%%% ERROR (Script Failed) | Line 1484 | Command: systemctl restart nginx | Exit code: 1 %%%

**Comment by ntsux** [comment]
I believe I have narrowed this done to using LetsEncrypt.  I believe this because everything works when I select the option to us a self-signed cert.

So, looking at my pre-flight for LetsEncrypt, I have the following set up in my environment:

1) a valid public a-record in use, dedicated for this project  (proven when I ping the the a-record my the OS of my guac server, I can see the external IP in the reply)

2) my local network firewall is allowing and  NATing TCP 443, TCP 80, TCP 8080, TCP 8443 to the internal IP of my guacamole server (not really sure that I need 8080/8443 allowed from the Internet, but allowing them anyway)

As per the script, the LetsEncrypt options I am using are:

i) a valid email address
ii) default keysize (4096)
iii) using OCSP (by default)

4) for Nginx, my selected options are:

i) using my local (real) IP of my guac server (default)
ii) using my valid public a-record
iii) URI path of "/" (default)
iv) 256-bit SSL ciphers (default)
v) using CSP (default)

...  I am still looking at this.  Perhaps it's not a script issue, but I don't see the issues in my pre-flight or the script LetsEncrypt or Niginx options (yet).

**Comment by ntsux** [comment]
Wireshark to the rescue.

This is not a bug with your script.  Hope we can can change the label.

Turns out an internal firewall in the test lab was dropping what it perceived to be a 'NAT reflection' for some outbound web calls that refer to the internal hosted service during the script execution.

We can kill this thread... or leave it up as a monument to my own embarrassing idiocy.

**Comment by Zer0CoolX** [comment]
Sorry I am just now seeing this, the world has been crazy lately. I am glad to see you got the issue worked out.

> *** By the way, this is one of the finest scripts I have ever seen, very well structured, with very well laid out documentation! Really appreciate your efforts here! ***

I appreciate the kind words. Hopefully it is evident I have put a lot of time and effort in the project. I have learned a great deal over the last few years writing this script and its a pleasure to be able to share the script and what I have learned about Guacamole as well.

If you like the script keep an eye out, I hope to post a revised RHEL/CentOS 8.x version with some substantial improvements. It will likely end up in its own repo but I will link to it from this one.

> We can kill this thread... or leave it up as a monument to my own embarrassing idiocy.

I will leave it as it could benefit others with similar issues. We all make mistakes, some of my commits are embarrassing, its better if no one goes back and looks at them lol.

**Comment by ntsux** [comment]
Sounds good, and I look forward to the CentOS 8.x version!

Thanks again!
On Jul. 14, 2020, at 08:39, Zer0CoolX <notifications@github.com<mailto:notifications@github.com>> wrote:

Sorry I am just now seeing this, the world has been crazy lately. I am glad to see you got the issue worked out.

*** By the way, this is one of the finest scripts I have ever seen, very well structured, with very well laid out documentation! Really appreciate your efforts here! ***

I appreciate the kind words. Hopefully it is evident I have put a lot of time and effort in the project. I have learned a great deal over the last few years writing this script and its a pleasure to be able to share the script and what I have learned about Guacamole as well.

If you like the script keep an eye out, I hope to post a revised RHEL/CentOS 8.x version with some substantial improvements. It will likely end up in its own repo but I will link to it from this one.

We can kill this thread... or leave it up as a monument to my own embarrassing idiocy.

I will leave it as it could benefit others with similar issues. We all make mistakes, some of my commits are embarrassing, its better if no one goes back and looks at them lol.
